app/lib/package/upload_signer_service.dart

```diff
@@ -28,6 +28,21 @@ UploadSignerService get uploadSigner =>
 void registerUploadSigner(UploadSignerService uploadSigner) =>
     ss.register(#_url_signer, uploadSigner);
 
+/// Creates an upload signer based on the current environment.
+Future<UploadSignerService> createUploadSigner(http.Client authClient) async {
```

Comment:
Should we change pattern and just call this: `setupUploadSigner` and have it call `registerUploadSigner` internally, instead of returning it?

Then we can make `registerUploadSigner` private.. perhaps that's one way to encapsulate things a bit.

Reply:
We also register the `FakeUploadSignerService` in the case of tests and fake pub server. As we move/integrate the fake pub server into the app project, I think we may have better patterns later on...